src/details/ArborX_Callbacks.hpp

```diff
@@ -172,6 +179,18 @@ KOKKOS_INLINE_FUNCTION
   return false;
 }
 
+template <typename Callback>
+using CallbackTreeTraversalControlArchetypeAlias =
+    typename Callback::TreeTraversalControl;
+
+template <typename Callback>
+struct callback_exact_distance
+    : std::is_same<KokkosExt::detected_t<
+                       CallbackTreeTraversalControlArchetypeAlias, Callback>,
+                   Experimental::ExactNearestDistance>
```

Comment:
Have you considered detecting the existence of the `distance` member function?

Reply:
Yes, this would need to be added, including the check that it returns `float`. This is simply a draft to get some agreement on a general direction.

Reply:
@dalg24 Please take a look. Now the presence of the distance function is detected, and there is no need to specify an extra typedef inside a callback.